### PR TITLE
Docs: Add admin docs page for the Outbound request log

### PR DESCRIPTION
**Pull request by vdavid**
This PR adds the missing admin docs for the new “Outbound request log” feature.

The URL is `https://docs.sourcegraph.com/admin/observability/outbound-request-log` (note the singular form)

## Test plan

- Checked for typos and grammar with Grammarly
- Added references to the new page from several places
- PR review

### CODE REVIEW

File: doc/admin/observability/outbound-request-log.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## Enabling the logs
+
+This feature is off by default. You can enable it by setting `outboundRequestLogLimit` to a non-zero value in the site config. The recommended value is `50`.
+
+## Privacy
+
+HTTP headers that are considered sensitive (such as `Authorization`) are redacted.
+The logs are stored in memory in Redis and are automatically cleared on an instance restart or when you disable the feature.
+
+## Logged data
```
Comment (jhchabran) [inline doc/admin/observability/outbound-request-log.md:20]: Perhaps you want to clarify that these are not going in the logs, and that we're talking about what's captured and stored instead.

Comment (vdavid) [reply doc/admin/observability/outbound-request-log.md:20]: Great point, thanks, JH!

File: doc/admin/observability/outbound-request-log.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Outbound request log
+
+There's a page in the web UI where instance admins can see all outbound requests generated by an instance. The page makes it easy to quickly see how Sourcegraph interacts with other systems, such as code hosts, without manually digging into logs, and is especially useful when debugging.
```
Comment (mrnugget) [inline doc/admin/observability/outbound-request-log.md:3]:
```suggestion
On the **Outbound requests** page site-admins can see all outbound requests generated by an instance. The page makes it easy to quickly see how Sourcegraph interacts with other systems, such as code hosts, without manually digging into logs, and is especially useful when debugging.
```

Let's name the page directly :) "There is a page" is confusing as the first sentence. Also: we call them site-admins (or "site admin", see: https://handbook.sourcegraph.com/company-info-and-process/communication/content_guidelines/terminology_guidelines/#term-usage)

File: doc/admin/observability/outbound-request-log.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+![UI screenshot](../img/outbound-request-log.png)
+
+## Prerequisites
+
+This document assumes you are a [site administrator](../index.md).
```
Comment (mrnugget) [inline doc/admin/observability/outbound-request-log.md:9]: See above: site admin.

File: doc/admin/observability/outbound-request-log.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+## Enabling the logs
+
+This feature is off by default. You can enable it by setting `outboundRequestLogLimit` to a non-zero value in the site config. The recommended value is `50`.
```
Comment (mrnugget) [inline doc/admin/observability/outbound-request-log.md:13]: Can you make `site config` a link to the docs page describing site config? See other examples: https://sourcegraph.sourcegraph.com/search?q=site+config+file%3Adoc&groupBy=repo

File: doc/admin/observability/outbound-request-log.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+You can later disable it by setting `outboundRequestLogLimit` to `0`, or by removing the setting entirely.
+
+## Privacy
+
+HTTP headers that may be sensitive (such as `Authorization`) are being redacted before saving. Sensitive information is not stored in the logs.
+
+Logs are stored in memory in Redis, and are automatically cleared on a Redis instance restart or when you disable the feature.
```
Comment (mrnugget) [inline doc/admin/observability/outbound-request-log.md:21]: I think that depends on how Redis is configured, right? If they use a single Redis instance configured to be used as cache and store, it won't.

File: doc/admin/observability/outbound-request-log.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+- Any errors
+- Request headers (with sensitive fields redacted)
+- Response headers (with sensitive fields redacted)
+- Request body
+
+## Copying the cURL command
+
+To recreate a request for debugging, you can copy the cURL command for any request by clicking the "Copy curl" button. Then you can paste the command into your terminal and run it, or paste it to something like [Postman](https://www.postman.com/) for further debugging.
+
+Keep in mind that some headers might be redacted (you'll see the word "REDACTED" in their place, in which case you'll need to add the missing pieces manually.
```
Comment (mrnugget) [inline doc/admin/observability/outbound-request-log.md:41]: parser error: expected ) got newline instead

**Review by mrnugget — APPROVED** [review]

File: doc/admin/observability/outbound-request-log.md (hunk 3 above)
Comment (vdavid) [reply doc/admin/observability/outbound-request-log.md:9]: Thanks. I copied this from another section of the docs. I've fixed this at several places in the docs now.

File: doc/admin/observability/outbound-request-log.md (hunk 4 above)
Comment (vdavid) [reply doc/admin/observability/outbound-request-log.md:13]: That's neat; I didn't know about this possibility.

File: doc/admin/observability/outbound-request-log.md (hunk 5 above)
Comment (vdavid) [reply doc/admin/observability/outbound-request-log.md:21]: Thanks, modified it to "Logs are stored in memory in Redis with an expiration of seven days. They are automatically cleared when you disable the feature."

File: doc/admin/observability/outbound-request-log.md (hunk 6 above)
Comment (vdavid) [reply doc/admin/observability/outbound-request-log.md:41]: Ah, great catch; I'm grateful for this one, none of my tools warned me about this mistake.

**Review by iskyOS — APPROVED** [review]
TY for adding!

**Pull request merged**